tiledb/api/c_api_support/exception_wrapper/exception_wrapper.h

```diff
@@ -561,27 +582,46 @@ class CAPIFunction<f, H> {
        * Note that we don't need std::forward here because all the arguments
        * must have "C" linkage.
        */
-      auto x{f(args...)};
-      h.action_on_success();
-      return x;
+      //-------------------------------------------------------
+      A::call(args...);
+      if constexpr (std::same_as<R, void>) {
+        f(args...);
+        h.action_on_success();
+      } else {
+        auto x{f(args...)};
+        h.action_on_success();
+        return x;
+      }
     } catch (const std::bad_alloc& e) {
       h.action(e);
-      return TILEDB_OOM;
+      if constexpr (!std::same_as<R, void>) {
```

Comment:
It's not worth it. It adds twenty (or more) lines of code to save ten. We need something like `return nonvoid_value<R>(TILEDB_OOM);`, which requires a generic definition, a specialization to `void` at minimum. And it's harder than it looks because `nonvoid_value` needs two template argument to handle the `void` case, since it needs to declare a type for its argument, even if it's going to ignore it.